### PR TITLE
feat(VSkeletonLoader): port to v3

### LINKED ISSUE #9194: [Bug Report] v-skeleton-loader (& v-boilerplate) custom types not rendered like pre-defined type
### Environment
**Vuetify Version:** 2.1.0
**Vue Version:** 2.6.10
**Browsers:** Waterfox 56.2.14
**OS:** Linux x86_64

### Steps to reproduce
Using a custom type (like `button@2`) does not render like its equivalent pre-defined type (`actions`).

### Expected Behavior
The custom type renders like its equivalent pre-defined type.

### Actual Behavior
Rendered differently.

### Reproduction Link
<a href="https://codepen.io/ibrahimbeladi/pen/xxKvmLz?editors=1010" target="_blank">https://codepen.io/ibrahimbeladi/pen/xxKvmLz?editors=1010</a>

### Other comments
Same goes to v-boilerplate (#9193).

### LINKED ISSUE #10172: [Feature Request] Remove width style from '.v-skeleton-loader__heading'
### Problem to solve
Currently, the "heading" type of `v-skeleton-loader` has a pre-set width within one of it's child classes (`.v-skeleton-loader__heading`). This width is set to 45%. 

With this width style present, it makes it very difficult to appropriately control the width of the skeleton using the `width` prop. Additionally, it makes it nearly impossible to get the skeleton to be the appropriate width AND centered within a container, as the skeleton is actually much larger than the displayed content. 

Codepen for reproduction:
https://codepen.io/Mattmilliman91/pen/wvBmPPw

### Proposed solution
Remove the offending width style to allow the `width` prop within the component to control the sizing of the skeleton. This will also address the centering issue.

Codepen for solution:
https://codepen.io/Mattmilliman91/pen/QWwmOVE

### LINKED ISSUE #10199: [Bug Report] VSkeletonLoader pre-defined table-row formatting issue
### Environment
**Vuetify Version:** 2.2.3
**Vue Version:** 2.6.11
**Browsers:** Chrome 80.0.3987.42
**OS:** Windows 10

### Steps to reproduce
1. Open the codepen and note that the two tables do not share matching css in the `table-row-divider` area.

### Expected Behavior
According to the [docs](https://vuetifyjs.com/en/components/skeleton-loaders#api), pre-defined types should match their non-pre-defined type equivalents.

### Actual Behavior
They do not match, as `table-tbody` != `table-row-divider@6` in the repro.

### Reproduction Link
<a href="https://codepen.io/185driver/pen/xxbzxJL" target="_blank">https://codepen.io/185driver/pen/xxbzxJL</a>

### Other comments
That they don't match becomes an issue if one wishes to use a different number of table-rows, or if the use of the `dark` theme is desired. Without the helper formatting of `table-tbody`, the css doesn't work as expected.

This is not the only pre-defined type to be unmatched to it's parts:
`table-heading` != `heading, text`
`table-thead` != `heading@6`
`table-row-divider` != `table-row, divider`
`table-row` != `table-cell@6`
`table-cell` != `text`
`table-tfoot` != `text@2, avatar@2`

So some of the `table-` types have sugar added, but others do not.

Related to #9194, but this issue relates to table types.

### LINKED ISSUE #13504: [Task] v-skeleton-loader
# Overview

- Related Tasks: n/a
- Related Issues: n/a

A conversion should not start until *all* assigned issues with the milestone v2.2.x are resolved and in the `next` branch. Reference our [Coding Guidelines](https://github.com/vuetifyjs/vuetify/wiki/coding-guidelines) for information regarding the team's practices or reach out to us on [Discord](https://community.vuetifyjs.com).

### PR DESCRIPTION
**Pull request by SteinRobert**
## Motivation
port skeleton loader to v3

fixes #9194
fixes #10199
fixes #10172
closes #13504

## Markup:

<details>

```vue
<template>
  <v-app>
    <div class="ma-4 pa-4">
      <v-card color="grey-lighten-4" max-width="900" class="pa-4">
        <v-skeleton-loader type="card" />
        <br>
        <br>
        <v-skeleton-loader type="avatar, button@2, text" />
        <br>
        <br>
        <v-skeleton-loader type="date-picker" />
        <br>
        <br>
        <v-skeleton-loader type="article" />
        <br>
        <br>
        <v-skeleton-loader type="button, chip" />
        <br>
        <br>
        <v-skeleton-loader type="text, button@2" />
        <br>
        <br>
        <v-skeleton-loader type="avatar, sentences" />
        <br>
        <br>
        <v-skeleton-loader type="avatar, paragraph" />
        <br>
        <br>
        <v-skeleton-loader type="avatar, text" />
        <br>
        <br>
        <v-skeleton-loader type="chip" />
        <br>
        <br>
        <v-skeleton-loader type="actions" />
        <br>
        <br>
        <v-skeleton-loader type="button@2" />
        <br>
        <br>
        <v-skeleton-loader type="heading, paragraph" />
        <br>
        <br>
        <v-skeleton-loader type="image, heading" />
        <br>
        <br>
        <v-skeleton-loader type="image, list-item-avatar" />
      </v-card>
    </div>
  </v-app>
</template>

<script setup>
  import { ref } from 'vue'
</script>
```
</details>

### CODE REVIEW

**Comment by johnleider** [comment]
An updated playground with some test cases put together:

```vue
<template>
  <v-app>
    <v-main>
      <v-responsive max-width="400">
        <div class="ma-4 pa-4 bg-grey-lighten-3">
          <v-skeleton-loader type="card" class="mb-4" />

          <div class="mb-4">
            <v-skeleton-bone type="image" />

            <v-skeleton-bone type="sheet">
              <v-skeleton-bone type="avatar" class="me-4" />

              <div class="flex-grow-1">
                <v-skeleton-bone height="16" type="heading" />
                <v-skeleton-bone height="8" type="subheading" />
              </div>
            </v-skeleton-bone>

            <v-skeleton-bone type="sheet">
              <v-skeleton-bone type="heading" />
            </v-skeleton-bone>

            <v-skeleton-bone type="sheet" class="d-block">
              <v-skeleton-bone type="text" />
              <v-skeleton-bone type="text" width="50%" />
              <v-skeleton-bone type="text" width="75%" />
            </v-skeleton-bone>

            <v-skeleton-bone type="sheet" class="justify-end">
              <v-skeleton-bone type="button" />
              <v-skeleton-bone type="button" />
            </v-skeleton-bone>
          </div>

          <div class="mb-4">
            <v-skeleton-bone type="sheet">
              <v-skeleton-bone type="heading" />
            </v-skeleton-bone>

            <v-skeleton-bone type="sheet" class="d-block">
              <v-skeleton-bone type="text" />
              <v-skeleton-bone type="text" width="50%" />
              <v-skeleton-bone type="text" width="75%" />
            </v-skeleton-bone>

            <v-skeleton-bone type="sheet">
              <v-spacer />
              <v-skeleton-bone type="button" />
              <v-skeleton-bone type="button" />
            </v-skeleton-bone>
          </div>

          <div>
            <v-skeleton-bone type="sheet">
              <v-skeleton-bone type="heading" width="36" />
              <v-spacer />
              <v-skeleton-bone type="subheading" />
            </v-skeleton-bone>

            <v-skeleton-bone type="sheet" class="d-block">
              <v-skeleton-bone type="text" />
              <v-skeleton-bone type="text" width="50%" />
            </v-skeleton-bone>

            <v-skeleton-bone type="image" />

            <v-skeleton-bone type="sheet" class="justify-space-between">
              <v-skeleton-bone type="subheading" />

              <v-skeleton-bone type="subheading" />

              <v-skeleton-bone type="avatar" />

              <v-skeleton-bone type="avatar" />
            </v-skeleton-bone>
          </div>
        </div>
      </v-responsive>
    </v-main>
  </v-app>
</template>

<script setup>
  import { ref } from 'vue'
</script>
```

File: packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  'table-thead': 'heading@6',
+  'table-tbody': 'table-row-divider@6',
+  'table-row-divider': 'table-row, divider',
+  'table-row': 'text@6',
+  'table-tfoot': 'text@2, avatar@2',
+  text: 'text',
+} as Record<string, string>
+
+function genBone (text: string, children: VNode[] = []) {
+  return h('div', {
```
Comment (nekosaur) [inline packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx:50]: any special reason for not using tsx?

File: packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  },
+
+  setup (props, { slots }) {
+    const { dimensionStyles } = useDimension(props)
+    const { elevationClasses } = useElevation(props)
+    const { backgroundColorClasses, backgroundColorStyles } = useBackgroundColor(toRef(props, 'color'))
+    const { themeClasses } = provideTheme(props)
+
+    const items = computed(() => {
+      return genStructure(props.type as any)
```
Comment (nekosaur) [inline packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx:120]: `any`

File: packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+function genBones (bone: string) {
+  // e.g. 'text@3'
+  const [type, length] = bone.split('@') as [string, number]
+
+  // Generate a length array based upon
+  // value after @ in the bone string
+  return Array.from({ length }).map(() => genStructure(type))
+}
+
+function genStructure (type?: string): any {
```
Comment (nekosaur) [inline packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx:67]: `any`

**Review by nekosaur — COMMENTED** [review]
Just a quick review of the component code so far.

File: packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx (hunk 1 above)
Comment (johnleider) [reply packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx:50]: Wasn't sure how to render the array of vnodes using it, open to suggestions

Comment (nekosaur) [reply packages/vuetify/src/labs/VSkeletonLoader/VSkeletonLoader.tsx:50]: Maybe I'm tired and not thinking straight, but shouldn't it just be
```js
<div class={[...]}>{ children }</div>
```

**Comment by yooneskh** [comment]
Can we have brs inside the type?

**Comment by johnleider** [comment]
I don't understand what you're asking

**Comment by yooneskh** [comment]
I might have understood worngly, but have can we have a skeleton loader for a layout that has an avatar with a text in front of the avatar and a paragraph below both?

**Comment by yooneskh** [comment]
Oh beautiful, got it. So each element is either inline or block. Sorry for asking. I thought we need to specify lines separately.

**Review by johnleider — APPROVED** [review]

**Pull request merged**